d2/src/main/java/com/linkedin/d2/balancer/properties/ClusterProperties.java

```diff
@@ -95,9 +102,17 @@ public ClusterProperties(String clusterName,
       DarkClusterConfigMap darkClusters)
   {
     this(clusterName, prioritizedSchemes, properties, bannedUris, partitionProperties, sslSessionValidationStrings,
-        darkClusters, false);
+        (Map<String, Object>)null, false);
   }
 
+  /**
+   * @deprecated Use the constructor that uses a Map instead of DarkClusterConfigMap. Using this object is not flexible enough to hold
+   * transportClientProperties, because {@link com.linkedin.d2.balancer.config.TransportClientPropertiesConverter} uses different
+   * keys in it's serialization than how Jackson would serialize D2TransportClientProperties. That is problematic since ClusterProperties
+   * already should have had all necessary conversions done, but in this case the pegasus objects don't have a mechanism to allow the conversions.
+   * The solution is to use a Map<String, Object> to pass in the darkClusters.
+   */
+  @Deprecated
```

Comment:
I don't think this is a good idea. It'd be confusing since the getter that jackson would use to serialize would have a Map<String, Object> but the constructor would have something different.